security/pkg/server/ca/server.go

```diff
@@ -52,10 +51,48 @@ type Server struct {
 	monitoring     monitoringMetrics
 }
 
+// CreateCertificate handles an incoming certificate signing request (CSR). It does
+// proper validation (e.g. authentication) and upon validated, signs a certificate that:
+// the SAN is the identity of the caller in authentication result.
+// the subject public key is the public key in the CSR.
+// it is signed by the CA signing key.
+func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertificateRequest) (
+	*pb.IstioCertificateResponse, error) {
+	caller := s.authenticate(ctx)
```

Comment:
just curious, ```authenticate``` calls a list of pre-configured grpc based authentication services for jwt authentication ? If yes, do we have some existing example authentication service ?

Reply:
Good to know, thanks.